sched/semaphore/sem_tickwait.c

```diff
@@ -165,18 +145,22 @@ int nxsem_tickwait(FAR sem_t *sem, clock_t start, uint32_t delay)
  *
  ****************************************************************************/
 
-int nxsem_tickwait_uninterruptible(FAR sem_t *sem, clock_t start,
-                                   uint32_t delay)
+int nxsem_tickwait_uninterruptible(FAR sem_t *sem, uint32_t delay)
 {
+  clock_t start = clock_systime_ticks();
   int ret;
 
   do
     {
       /* Take the semaphore (perhaps waiting) */
 
-      ret = nxsem_tickwait(sem, start, delay);
+      ret = nxsem_tickwait(sem, delay);
```

Comment:
this code is bad. `delay` is not adjusted in case if `nxsem_tickwait` was interrupted, so for example in case of `nxsem_tickwait_uninterruptible(sem, 10);` the `nxsem_tickwait` gets interrupted at 9th tick, the `clock_systime_ticks() - start` == `9` and `9 < 10` so `nxsem_tickwait(sem, 10);` will be called the second time and will lead to `19` ticks delay instead of `10`.
The existing code was exactly handing such cases correctly.

Reply:
Yes, the last wait is longer than expect, let me modify it.

Reply:
@pkarashchenko Done, please review again.